text/0060-bazel.md

```diff
@@ -0,0 +1,137 @@
+yarn.lock
+```
+
+The `BUILD` files themselves are quite simple, since Bazel has rules built-in for compiling a TypeScript library and for
+running JavaScript tests. We would need to write custom rules in the 
+[Skylark language](https://docs.bazel.build/versions/master/skylark/language.html) to perform custom operations like
+building TypeScript definitions from CloudFormation definition files. This is relatively straightforward since the tooling
+for this exists in TypeScript already, and Skylark is a subset of Python.
+
+Example `BUILD.bazel` file, assuming `cdk_library` is a Bazel rule that calls `ts_library` with another rule that performs 
+TS transformation:
+```
+load("//tools:defaults.bzl", "cdk_library", "pkg_npm")
+
+package(default_visibility = ["//visibility:public"])
+
+cdk_library(
+    name = "aws-amazonmq",
+    scope = ["AWS::AmazonMQ"],
+    outputs = ["lib/amazonmq.generated.ts"],
+    srcs = glob(["lib/**/*.ts"]),
+    deps = [
+        "//packages/@aws-cdk/cdk",
+    ],
+)
+
+pkg_npm(
+    name = "npm_package",
+    srcs = [
+      "package.json",
+    ],
+    tags = [
+        "release-with-framework",
+    ],
+    deps = [
+        ":aws-amazonmq",
+    ],
+)
+```
+
+The `pkg_npm` rule is also built-in to Bazel, and bundles a package based on given sources, along with other needed files like a package.json.
+Note that the `cdk_library` rule is only needed for packages that have a CFN to TS transformation. If that's not needed, it's simply a `ts_library` rule.
+
+# Drawbacks
+
+There are two main drawbacks to migrating to Bazel:
+
+1. Lack of familiarity with Bazel amongst the team
+1. Relative newness of Bazel in the JavaScript/TypeScript ecosystem
+
+For the first point, the current system has the advantage of being a glorified monorepo wrapper around NPM. Bazel,
+on the other hand, has a completely different semantic structure. While using it is straightforward for the most part,
+understanding how it works and more importantly how to extend it (by writing custom rules, for example) is not.
+
+For the second point, while Bazel has had its 1.0 release, that release was relatively recent, and there is still much
+work to be done in the space. For instance, Bazel does not currently supported nested package.json dependencies when
+constructing its own dependency tree. While the nested files can exist, their dependencies would have to be deduped to
+the root package.json file in order for everything to be setup properly by Bazel.
+
+# Rationale and Alternatives
+
+## Rationale
+* Bazel is incremental, meaning that only the changed aspects of the project and its dependencies need to be rebuilt
+* Bazel caches all of its builds, making it much faster than the current solution
+* Bazel allows for hermiticity and reproducability since it runs in a sandbox
+* Bazel is highly extensible, meaning custom rules can be written solely in TypeScript/JavaScript as opposed to Bash, but
+still incorporated easily into the toolchain
```

Comment:
A drawback that has not been mentioned earlier is that Bazel introduces yet another dependency in the mix that you need to install (and later maintain) in order to contribute to CDK. I believe it can be installed from `npm` (mitigating this problem to a large extent), but I think it needs to be mentioned in the document.

Reply:
It can be installed from NPM, and in that way doesn't introduce that much overhead. But definitely worth a mention, you're totally right.